lib/cog/commands/echo.ex

```diff
@@ -8,5 +8,15 @@ defmodule Cog.Commands.Echo do
   rule "when command is #{Cog.Util.Misc.embedded_bundle}:echo allow"
 
   def handle_message(req, state),
-    do: {:reply, req.reply_to, Enum.join(req.args, " "), state}
+    do: {:reply, req.reply_to, Enum.map_join(req.args, " ", &serialize/1), state}
+
+  # We should only have to worry about maps and lists here. Anything else
+  # we might get should implement 'to_string'.
+  defp serialize(map) when is_map(map),
+    do: Poison.encode!(map)
+  defp serialize(list) when is_list(list),
+    do: Poison.encode!(list)
+  defp serialize(val),
+    do: "#{val}"
```

Comment:
I'd just call `to_string(val)` directly, instead of doing an unnecessary string interpolation.